src/internal/observable/dom/animationFrames.ts

```diff
@@ -72,25 +72,29 @@ export interface TimestampProvider {
  * const source$ = animationFrames(customTSProvider);
  *
  * // Log increasing numbers 0...1...2... on every animation frame.
- * source$.subscribe(x => console.log(x));
+ * source$.subscribe(({elapsed}) => console.log(elapsed));
  * ```
  *
  * @param timestampProvider An object with a `now` method that provides a numeric timestamp
  */
-export function animationFrames(timestampProvider: TimestampProvider = Date) {
-  return timestampProvider === Date ? DEFAULT_ANIMATION_FRAMES : animationFramesFactory(timestampProvider);
+export function animationFrames(timestampProvider?: TimestampProvider) {
+  return timestampProvider ? animationFramesFactory(timestampProvider) : DEFAULT_ANIMATION_FRAMES;
 }
 
 /**
  * Does the work of creating the observable for `animationFrames`.
  * @param timestampProvider The timestamp provider to use to create the observable
  */
-function animationFramesFactory(timestampProvider: TimestampProvider) {
-  return new Observable<number>(subscriber => {
+function animationFramesFactory(timestampProvider?: TimestampProvider) {
+  return new Observable<{timestamp: number, elapsed: number}>(subscriber => {
     let id: number;
-    const start = timestampProvider.now();
-    const run = () => {
-      subscriber.next(timestampProvider.now() - start);
+    let start: DOMHighResTimeStamp | number | null = null;
+    const run = (timestamp: DOMHighResTimeStamp | number) => {
+      const currentTimestamp = timestampProvider ? timestampProvider.now() : timestamp;
+      if (start === null) {
+        start = currentTimestamp;
```

Comment:
Okay, the only issue I have with this now is that the `start` time is now non-deterministic. I think collection of `start` should be moved back to where it was. This is no longer the "start" it's the first animation frame.

Consider the following scenario: A user triggers an animation that is supposed to take 30 seconds, then, before the animation frame fires they switch tabs. If the start isn't collected until the first animationFrame, when the user tabs back, they'll be forced to watch the full animation.

If the `start` is collected when the observable is subscribed to, when they come back, the animation will be done as expected.

I understand that maybe this change probably arose from the confusion about it being a DOMHighResTimeStamp, and the code was previously using `Date` as default timestamp provider. Instead, it's safe to use `performance` as the default TimestampProvider, as it is supported by all browsers and platforms we currently support.